### PR TITLE
Add github actions to aid the release process

### PR DESCRIPTION
**Pull request by shaynafinocchiaro**
#### Is this a new chart?

No

#### What this PR does / why we need it:
This PR adds 2 additional github actions
- Post Release Action updates references of target-branch, mostly for dependabot
- Sync Release Branch with Main will try to rebase the feature branch with main when there are pushes to main branch. If a rebase can't be performed automatically, an issue is created and assigned to the person who last merged to main. This is done in hopes to ease the release of the helm-charts.

#### Which issue(s) is this PR associated with:

- https://github.com/dell/csm/issues/1490

#### Checklist:

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [ ] Chart Version bumped
- [ ] Variables are documented in the chart README.md
- [ ] Title of the PR starts with the chart name (e.g. `[charts_dir/mychartname]`) if applicable
Screenshot 2024-11-07 093222
Screenshot 2024-11-07 093306

### CODE REVIEW

**Review by alikdell — DISMISSED** [review]

**Review by gallacher — APPROVED** [review]

File: .github/workflows/post-release-action.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v7
+        with:
+          token: ${{ steps.generate-token.outputs.token }}
+          branch: 'update-target-branch'
+          title: 'Update target-branch references for dependabot'
+          body: |
+            This PR updates target-branch references to the new release branch.
+            Auto-generated by [dell/helm-charts](https://github.com/dell/helm-charts)
```
Comment (ChristianAtDell) [inline .github/workflows/post-release-action.yml:48]: This seems very repo-specific; do our release branch actions need to be manually done in each repository, or should they be common github actions? Can we generalize them up to that level?

File: .github/workflows/post-release-action.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+      # Needed for signing commits using Github App tokens
+      # See: https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#commit-signing
+      - uses: actions/create-github-app-token@v1.11.0
+        id: generate-token
+        with:
+          app-id: ${{ vars.CSM_RELEASE_APP_ID }}
+          private-key: ${{ secrets.CSM_RELEASE_APP_PRIVATE_KEY }}
+
+      - name: Create Pull Request
```
Comment (ChristianAtDell) [inline .github/workflows/post-release-action.yml:40]: This action is titled 'Create Pull Request', is it just a general action to create a pull request or is it specific to the release branch? If it's a general action, does it have any inputs that can be used to create PRs for specific needs?

**Review by ChristianAtDell — DISMISSED** [review]
This is an impressive bit of work that could make releases faster/more automated. Has it been tested in a private clone of the repository to make sure it creates the proper branch? Does it do any tagging? I didn't see actions for that but I figure that's the next logical step.

**Review by atye — APPROVED** [review]

File: .github/workflows/post-release-action.yml (hunk 1 above)
Comment (shaynafinocchiaro) [reply .github/workflows/post-release-action.yml:48]: I'm only aware of this repo needing to target a specific release branch for dependabot that's why I didn't put this in common-github-actions repo

File: .github/workflows/post-release-action.yml (hunk 2 above)
Comment (shaynafinocchiaro) [reply .github/workflows/post-release-action.yml:40]: The PR is specific to the file changes that come from the step `Update branch references`

Comment (ChristianAtDell) [reply .github/workflows/post-release-action.yml:40]: Should we rename the action to something that's more specific, then? 'Create PR for Release Updates' or similar?

Comment (shaynafinocchiaro) [reply .github/workflows/post-release-action.yml]: Fixed

**Review by ChristianAtDell — APPROVED** [review]

**Review by AronAtDell — APPROVED** [review]

**Review by tdawe — APPROVED** [review]
LGTM

**Pull request merged**